esp-hal/src/spi/master.rs

```diff
@@ -2591,13 +2611,13 @@ impl Driver {
     ) -> Result<(), Error> {
         let reg_block = self.register_block();
         match cmd_mode {
-            DataMode::Single => (),
+            DataMode::SingleTwoDataLines => (),
```

Comment:
If you know of a limitation why we reject Single here, please add it as a todo or a note. I don't think we read during the command phase, and the "normal" half duplex modes make me say we would instead want `Single` here, if not both.

Reply:
this one?

![image](https://github.com/user-attachments/assets/f4deb616-8bb0-4343-b936-4a3f9a4b0fd6)

probably, you know better?

Reply:
Yeah we can't use Dual, Quad, etc - but nothing seems to indicate to me that `Single` wouldn't work, either. Right now, is `Single` even useable with ESP32? The QA tests all use `SingleTwoDataLines` which is I think pretty strange, as the device we talk to shouldn't really respond during the command and address phases, so maybe we should actually test both data modes and also accept both of them here.

Reply:
ok - took me a bit too long to realize what you were talking about at all

Reply:
and yes - I somehow ignored ESP32 before

Reply:
added _some_ testing of `Single` but in general I'm not too convinced by the half-duplex tests in general - maybe we can come up with something that actually can really test reading data

Reply:
For now I'm happy with the "it does not panic" guarantee we get.